client/src/app/project-create-edit/pages/create-project/create-project.component.ts

```diff
@@ -0,0 +1,102 @@
+import {Component, OnDestroy, OnInit} from '@angular/core';
+import {ActivatedRoute} from '@angular/router';
+import {AutofillProjectService} from '../../services/autofill-project.service';
+import {ProjectCacheService} from '../../services/project-cache.service';
+import {Identifier} from '../../models/europe-pmc-search';
+import {Observable, of, Subject} from 'rxjs';
+import {delay, map, takeUntil} from 'rxjs/operators';
+import {AutofillProject} from '../../models/autofill-project';
+import {IngestService} from '../../../shared/services/ingest.service';
+import {environment} from "../../../../environments/environment";
+
+const EMPTY_PROJECT = {
+  content: {},
+  isInCatalogue: true,
+};
+
+@Component({
+  selector: 'app-create-project',
+  templateUrl: './create-project.component.html',
+  styleUrls: ['./create-project.component.css']
+})
+export class CreateProjectComponent implements OnInit, OnDestroy {
+
+  project$: Observable<any>;
+  userIsWrangler: boolean;
+
+  private unsubscribe = new Subject<void>();
+
+  constructor(private route: ActivatedRoute,
+              private autofillProjectService: AutofillProjectService,
+              private projectCacheService: ProjectCacheService,
+              private ingestService: IngestService,
+  ) { }
+
+  ngOnInit(): void {
+    const queryParams = this.route.snapshot.queryParamMap;
+
+    if (queryParams.has(Identifier.DOI)) {
+      this.project$ = this.autofillProjectDetails(Identifier.DOI, queryParams.get(Identifier.DOI));
+    } else if (queryParams.has('restore')) {
+      this.project$ = this.projectCacheService.getProject();
+    } else {
+      this.project$ = of(EMPTY_PROJECT);
+    }
+
+    this.ingestService.getUserAccount()
+      .subscribe((account) => {
+        this.userIsWrangler = account.isWrangler();
+      });
+  }
+
+  ngOnDestroy() {
+    this.unsubscribe.next();
+  }
+
+  private autofillProjectDetails(id, search): Observable<object> {
```

Comment:
Wouldn't this conversion function need a test?

Reply:
There was no test  when it was implemented previously that I could find. That isn't to say I don't think there should be one just that I wasn't concerned with adding new testing in this PR